hibernate-core/src/main/java/org/hibernate/type/descriptor/java/JdbcTimestampJavaType.java

```diff
@@ -224,16 +223,8 @@ public void appendEncodedString(SqlAppender sb, Date value) {
 	@Override
 	public Date fromEncodedString(CharSequence charSequence, int start, int end) {
 		try {
-			final TemporalAccessor accessor = DateTimeUtils.DATE_TIME.parse( subSequence( charSequence, start, end ) );
-			final Timestamp timestamp;
-			if ( accessor.isSupported( ChronoField.INSTANT_SECONDS ) ) {
-				timestamp = new Timestamp( accessor.getLong( ChronoField.INSTANT_SECONDS ) * 1000L );
-				timestamp.setNanos( accessor.get( ChronoField.NANO_OF_SECOND ) );
-			}
-			else {
-				timestamp = Timestamp.valueOf( LocalDateTime.from( accessor ) );
-			}
-			return timestamp;
+			final TemporalAccessor accessor = ENCODED_FORMATTER.parse( subSequence( charSequence, start, end ) );
+			return Timestamp.from( Instant.from( accessor ) );
```

Comment:
This looks fine to me, this is slightly different from `JdbcTimeJavaType` and `JdbcDateJavaType` which use the `LITERAL_FORMATTER` in `appendEncodedString`, but is in-line with them by using `ENCODED_FORMATTER` when reading with `fromEncodedString`.

Only a minor concern: the check for support of `ChronoField.INSTANT_SECONDS` might be needed in some cases, so please restore that (also avoids instantiating an `java.time.Instant` just to then convert to `java.sql.Timestamp`)
```suggestion
			final TemporalAccessor accessor = ENCODED_FORMATTER.parse( subSequence( charSequence, start, end ) );
			final Timestamp timestamp;
			if ( accessor.isSupported( ChronoField.INSTANT_SECONDS ) ) {
				timestamp = new Timestamp( accessor.getLong( ChronoField.INSTANT_SECONDS ) * 1000L );
				timestamp.setNanos( accessor.get( ChronoField.NANO_OF_SECOND ) );
			}
			else {
				timestamp = Timestamp.valueOf( LocalDateTime.from( accessor ) );
			}
			return timestamp;
```